Assets/MRTK/Core/Interfaces/EventSystem/IMixedRealityEventPropagationData.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine.EventSystems;
+
+namespace Microsoft.MixedReality.Toolkit
+{
+    /// <summary>
+    /// Direction of event propagation through a hierarchy of elements
+    /// </summary>
+    [System.Flags]
+    public enum EventPropagation
```

Comment:
Having not read the full PR but reading just the interfaces, it seems we may not have a way to [cancel an event](https://www.w3.org/TR/DOM-Level-2-Events/events.html#Events-flow-cancelation). Is there a way to accomplish this? I think it would be good to try and match the DOM API here.

There is a good example of why we would need cancellation in the DOM spec:

> An example of this is a hyperlink in a web browser. When the user clicks on the hyperlink the default action is generally to active that hyperlink. Before processing these events, the implementation must check for event listeners registered to receive the event and dispatch the event to those listeners. These listeners then have the option of canceling the implementation's default action or allowing the default action to proceed. In the case of the hyperlink in the browser, canceling the action would have the result of not activating the hyperlink.

Reply:
Yeah you are right Julia, cancellation would still be possible by checking if event was used. But a more clear cancellation api would make sense by adding OnCancel events or even adding a Cancellation phase to the propagation.